src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -259,7 +263,31 @@ private static string FindBestVersionMatch(VersionRange versionRange, IEnumerabl
                 return version;
             }).Where(v => v != null);
 
-            return bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault()?.ToString();
+            var matchingVersion = bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();
+
+            if (bundleId != ScriptConstants.DefaultExtensionBundleId)
+            {
+                return matchingVersion?.ToString();
+            }
+
+            // Check to see if there is a max bundle version set via hosting configuration, if yes then use that instead of the one
+            // available on VM or local machine
+            if (!string.IsNullOrEmpty(configOption.MaximumBundleV3Version))
+            {
+                var maximumBundleV3Version = NuGetVersion.Parse(configOption.MaximumBundleV3Version);
+                matchingVersion = matchingVersion?.Major == ScriptConstants.ExtensionBundleV3MajorVersion && matchingVersion > maximumBundleV3Version
+                                ? maximumBundleV3Version
+                                : matchingVersion;
+            }
+            if (!string.IsNullOrEmpty(configOption.MaximumBundleV4Version))
```

Comment:
Is it possible that the HostingConfig has value for both V3 and V4? Do we need to execute both blocks here (v3 and v4)? Wondering it makes sense to return at like 280 if we got a new matchingVersion.

Reply:
Yes, We can return on line 280. We do match major versions before use the max configured version. Going to make that change.